PWGHF/Tasks/HFMCValidation.cxx

```diff
@@ -137,61 +171,132 @@ struct ValidationGenLevel {
     registry.fill(HIST("hCounterPerCollisionDplus"), counter[0]);
     registry.fill(HIST("hCounterPerCollisionDstar"), counter[1]);
     registry.fill(HIST("hCounterPerCollisionDzero"), counter[2]);
-    registry.fill(HIST("hCounterPerCollisionLambdaC"), counter[3]);
+    registry.fill(HIST("hCounterPerCollisionDs"), counter[3]);
+    registry.fill(HIST("hCounterPerCollisionLambdaC"), counter[4]);
+    registry.fill(HIST("hCounterPerCollisionXiC"), counter[5]);
+    registry.fill(HIST("hCounterPerCollisionJPsi"), counter[6]);
   }
 };
 
 /// Rec Level Validation
 ///
-/// Only D0 matched candidates:
+/// D±      → π± K∓ π±
+/// Ds±     → K± K∓ π±,
+/// D0(bar) → π± K∓,
+/// Λc±     → p(bar) K∓ π±
+/// Ξc±     → p(bar) K∓ π±
+/// J/psi   → e∓ e±
 ///   - Gen-Rec Level Momentum Difference per component;
 ///   - Gen-Rec Level Difference for secondary Vertex coordinates and decay length;
 struct ValidationRecLevel {
-  HistogramRegistry registry{
-    "registry",
-    {{"histPt", "Pt difference reco - MC; #it{p}_{T}^{reco} - #it{p}_{T}^{gen.} (GeV/#it{c}); entries", {HistType::kTH1F, {{2000, -1, 1}}}},
-     {"histPx", "Px difference reco - MC; #it{p}_{x}^{reco} - #it{p}_{x}^{gen.} (GeV/#it{c}); entries", {HistType::kTH1F, {{2000, -1, 1}}}},
-     {"histPy", "Py difference reco - MC; #it{p}_{y}^{reco} - #it{p}_{y}^{gen.} (GeV/#it{c}); entries", {HistType::kTH1F, {{2000, -1, 1}}}},
-     {"histPz", "Pz difference reco - MC; #it{p}_{z}^{reco} - #it{p}_{z}^{gen.} (GeV/#it{c}); entries", {HistType::kTH1F, {{2000, -1, 1}}}},
-     {"histSecVx", "Sec. Vertex difference reco - MC (MC matched); #Delta x (cm); entries", {HistType::kTH1F, {{200, -1, 1}}}},
-     {"histSecVy", "Sec. Vertex difference reco - MC (MC matched); #Delta y (cm); entries", {HistType::kTH1F, {{200, -1, 1}}}},
-     {"histSecVz", "Sec. Vertex difference reco - MC (MC matched); #Delta z (cm); entries", {HistType::kTH1F, {{200, -1, 1}}}},
-     {"histDecLen", "Decay Length difference reco - MC (MC matched); #Delta L (cm); entries", {HistType::kTH1F, {{200, -1, 1}}}}}};
 
-  Configurable<int> d_selectionFlagD0{"d_selectionFlagD0", 0, "Selection Flag for D0"};
-  Configurable<int> d_selectionFlagD0bar{"d_selectionFlagD0bar", 0, "Selection Flag for D0bar"};
+  static const int nCharmHad = 7;
+  std::array<std::shared_ptr<TH1>, nCharmHad> histPt, histPx, histPy, histPz, histSecVx, histSecVy, histSecVz, histDecLen;
+
+  HistogramRegistry registry{"registry", {}};
+  void init(o2::framework::InitContext&)
+  {
+    for (auto iHad = 0; iHad < nCharmHad; ++iHad) {
+      histPt[iHad] = registry.add<TH1>(Form("histPt%s", particleNames[iHad].data()), Form("Pt difference reco - MC %s; #it{p}_{T}^{reco} - #it{p}_{T}^{gen.} (GeV/#it{c}); entries", labels[iHad].data()), HistType::kTH1F, {{2000, -1, 1}});
```

Comment:
gen.->gen (everywhere)

Reply:
@vkucera @fcolamar if you agree I would postpone this discussion to a following PR, so if you agree with the modifications would it be possible to merge this one? Thanks!